src/algorithms/semihdp_sampler.hpp

```diff
@@ -0,0 +1,205 @@
+#ifndef SRC_ALGORITHMS_SEMI_HDP_SAMPLER_HPP
+#define SRC_ALGORITHMS_SEMI_HDP_SAMPLER_HPP
+
+#include <omp.h>
+#include <proto/cpp/semihdp.pb.h>
+
+#include <Eigen/Dense>
+#include <numeric>
+#include <src/collectors/memory_collector.hpp>
+#include <src/hierarchies/base_hierarchy.hpp>
+#include <src/utils/distributions.hpp>
+#include <src/utils/rng.hpp>
+#include <stan/math/prim.hpp>
+#include <stdexcept>
+#include <vector>
+
+using bayesmix::MarginalState;
+using bayesmix::SemiHdpState;
+using Eigen::MatrixXd;
+using Eigen::VectorXd;
```

Comment:
We should use this when there's really long class names involved, and this is not the case. I will change them myself in a future commit if you can't be bothered, I completely understand :eyes: